core/tauri/src/window.rs

```diff
@@ -2228,15 +2228,15 @@ impl<R: Runtime> Window<R> {
     window_label: Option<String>,
     event: String,
     handler: CallbackFn,
-  ) -> crate::Result<usize> {
-    let event_id = rand::random();
+  ) -> crate::Result<EventId> {
+    let event_id = self.manager.listeners().next_event_id();
 
     self.eval(&crate::event::listen_js(
-      self.manager().event_listeners_object_name(),
-      format!("'{}'", event),
+      self.manager().listeners().function_name(),
```

Comment:
this one was broken, changed listeners_object_name to function_name messed up the evaluated JS. I'm pushing a fix